src/PrestaShopBundle/Service/DataProvider/Admin/CategoriesProviderBuilder.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Service\DataProvider\Admin;
+
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use PrestaShop\PrestaShop\Core\Addon\Theme\ThemeRepository;
+
+/**
+ * Provide the categories used to order modules and themes on https://addons.prestashop.com.
+ */
+class CategoriesProviderBuilder
+{
+    private $context;
+    private $themeRepository;
+
+    private $addonsCategories;
+
+    public function __construct(LegacyContext $context, ThemeRepository $themeRepository, array $addonsCategories)
+    {
+        $this->context = $context;
+        $this->themeRepository = $themeRepository;
+        $this->addonsCategories = $addonsCategories;
+    }
+
+    public function build()
+    {
+        $themeName = $this->context->getContext()->shop->theme_name;
```

Comment:
If you need to work with legacy code you must do it inside `Adapter` namespace (sorry 😅 we have not yet discussed this as part of your onboarding). This is why PHPStan is not happy.

Reply:
Ok I move it to the adapter side :smiley: